casaos-healthcheck/run.sh

```
@@ -140,23 +140,152 @@ check_dns_resolution() {
+    print_header "System Temperature Check"
+    if command -v sensors &> /dev/null; then
+        sensors | grep -E 'Core|temp' | while read -r line; do
+            local temp=$(echo "$line" | awk '{print $2}' | sed 's/+//' | sed 's/°C//')
+            if (( $(awk "BEGIN {print ($temp > 80) ? 1 : 0}") )); then
+                print_color "0;31" "${CROSS_MARK} High temperature detected: ${line}"
+            else
+                print_color "0;32" "${CHECK_MARK} Temperature is normal: ${line}"
+            fi
+        done
+    elif [ -f "/sys/class/thermal/thermal_zone0/temp" ]; then
+        local temp=$(awk '{print $1/1000}' /sys/class/thermal/thermal_zone0/temp)
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)